### PR TITLE
ENH: validates boolean kwargs in DataFrame and Series methods

### LINKED ISSUE #16714: inplace kwarg must be of bool type, but other boolean kwargs don't have this restriction
#### Code Sample, a copy-pastable example if possible

Using `DataFrame.set_index` as an example, but this applies to other methods that take an inplace kwarg:

```python
# The following work
df.set_index('foo', drop=0)
df.set_index('foo', append='yes please')
df.set_index('foo', verify_integrity=22.7)
# This doesn't
df.set_index('foo', inplace=1)
```
The last line raises `ValueError: For argument "inplace" expected type bool, received type int.1`

#### Problem description

This behaviour seems inconsistent.

#### Expected Output

The last line should proceed successfully with inplace treated as `True`.

Or every boolean kwarg should be type-checked, though I like this option much less. I think using 0/1 as shorthand for True/False is a pretty common idiom.

It looks like this is the result of a somewhat recent 'bug fix': #14189. To me, it feels like a regression.

#### Output of ``pd.show_versions()``

<details>
INSTALLED VERSIONS
------------------
commit: None
python: 2.7.12.final.0
python-bits: 64
OS: Linux
OS-release: 4.4.0-78-generic
machine: x86_64
processor: x86_64
byteorder: little
LC_ALL: None
LANG: en_US.utf8
LOCALE: en_US.UTF-8

pandas: 0.20.1
pytest: None
pip: 9.0.1
setuptools: 36.0.1
Cython: None
numpy: 1.12.1
scipy: 0.19.0
xarray: None
IPython: 5.4.1
sphinx: None
patsy: None
dateutil: 2.6.0
pytz: 2017.2
blosc: None
bottleneck: None
tables: 3.3.0
numexpr: 2.6.2
feather: None
matplotlib: 2.0.2
openpyxl: None
xlrd: None
xlwt: None
xlsxwriter: None
lxml: None
bs4: 4.4.1
html5lib: 0.9999999
sqlalchemy: None
pymysql: None
psycopg2: None
jinja2: 2.9.6
s3fs: None
pandas_gbq: None
pandas_datareader: None
</details>

### PR DESCRIPTION
**Pull request by Condielj**
- [X] closes #16714 (for DataFrame and Series)
- [X] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [X] pre commit code check passed
- [X] Added an entry in the latest whatsnew.

(First commit is an extra frame.py and the second is deleting it- all frame.py's are in the right place)

### CODE REVIEW

File: doc/source/whatsnew/v1.5.0.rst [hunk 1]
```diff
@@ -125,6 +125,7 @@ Other enhancements
 - Added ``validate`` argument to :meth:`DataFrame.join` (:issue:`46622`)
 - A :class:`errors.PerformanceWarning` is now thrown when using ``string[pyarrow]`` dtype with methods that don't dispatch to ``pyarrow.compute`` methods (:issue:`42613`)
 - Added ``numeric_only`` argument to :meth:`Resampler.sum`, :meth:`Resampler.prod`, :meth:`Resampler.min`, :meth:`Resampler.max`, :meth:`Resampler.first`, and :meth:`Resampler.last` (:issue:`46442`)
+- Added validation of boolean kwargs in string series and DataFrame methods, along with tests (:issue:`16714`)
```
Comment (jreback) [inline doc/source/whatsnew/v1.5.0.rst:128]: pls be very specific on which methods / kwargs are updated here a reader is not going to find this note useful.

Comment (jreback) [reply doc/source/whatsnew/v1.5.0.rst:128]: i would also put this in the appropriate bug fix categories (even if you do multiple notes is ok)

File: pandas/util/_validators.py [hunk 2]
```diff
@@ -518,3 +520,39 @@ def validate_insert_loc(loc: int, length: int) -> int:
     if not 0 <= loc <= length:
         raise IndexError(f"loc must be an integer between -{length} and {length}")
     return loc
+
+def validate_bool_kwargs_from_keywords(*keywords):
```
Comment (jreback) [inline pandas/util/_validators.py:524]: umm why cannot we juse call the existng `validate_bool_kwarg`? why make yet another function?

**Review by jreback — CHANGES_REQUESTED** [review]
ideally do this incrementally to make sure tests pass

**Comment by mroeschke** [comment]
Thanks for the pull request, but it appears to have gone stale. If interested in continuing, please merge in the main branch, address any review comments and/or failing tests, and we can reopen.

**Pull request closed without merging**